Source/JavaScriptCore/runtime/JSGlobalObjectFunctions.cpp

```diff
@@ -476,20 +476,31 @@ JSC_DEFINE_HOST_FUNCTION(globalFuncEval, (JSGlobalObject* globalObject, CallFram
     auto scope = DECLARE_THROW_SCOPE(vm);
 
     JSValue x = callFrame->argument(0);
-    if (!x.isString())
+    JSString* codeString = nullptr;
+    if (x.isString())
+        codeString = asString(x);
+    else if (x.isObject()) {
+        auto code = globalObject->globalObjectMethodTable()->codeForEval(globalObject, x);
+        if (code.isNull())
+            return JSValue::encode(x);
+        codeString = JSC::jsString(vm, code);
+    } else
         return JSValue::encode(x);
 
+    String s = codeString->value(globalObject);
+    RETURN_IF_EXCEPTION(scope, encodedJSValue());
+
+    if (!globalObject->globalObjectMethodTable()->canCompileStrings(globalObject, CompilationType::IndirectEval, s, x)) {
```

Comment:
Add a boolean to the global object to indicate TT enforcement (report only or fully enforced), and only then do the call through to webcore.

Reply:
Do the same to the other file